ui/apps/platform/cypress/integration/clusters/cluster-registration-secrets/ClusterRegistrationSecrets.test.ts

```diff
@@ -0,0 +1,60 @@
+import withAuth from '../../../helpers/basicAuth';
+import { readFileFromDownloads } from '../../../helpers/file';
+import { getInputByLabel } from '../../../helpers/formHelpers';
+
+// TODO The entry point for this functionality will change once the removal from integrations is complete
+import { visitIntegrationsDashboardFromLeftNav } from '../../integrations/integrations.helpers';
+
+import { cleanupClusterRegistrationSecretsWithName } from './ClusterRegistrationSecrets.helpers';
+
+describe('Cluster registration secrets', () => {
+    withAuth();
+
+    const testCrsName = 'CYPRESS-E2E-TEST-CRS';
+
+    beforeEach(() => {
+        cleanupClusterRegistrationSecretsWithName(testCrsName);
+    });
+
+    afterEach(() => {
+        cleanupClusterRegistrationSecretsWithName(testCrsName);
+    });
+
+    it('should create a new Cluster registration secret and then view and delete', () => {
+        visitIntegrationsDashboardFromLeftNav();
+
+        cy.get('section:contains("Authentication Tokens")').scrollIntoView();
+        cy.get(
+            'section:contains("Authentication Tokens") a:contains("Cluster Registration Secret")'
+        ).click();
+
+        const crsLinkInTableSelector = `td[data-label="Name"] a:contains("${testCrsName}")`;
+
+        // Verify that old CRS from Cypress tests are not present
+        cy.get('table');
+        cy.get(crsLinkInTableSelector).should('not.exist');
+
+        // Create a new CRS and verify that the YAML is downloaded and the secret appears in the table
+        // TODO dv 2025-04-01
+        // From a user's point of view, this is a "button". It would be nice if we had a unified way to
+        // click a button without needing to know the underlying HTML element used by the component.
```

Comment:
Yeah, that's a fair point. If this button ends up changing internally, we will have to update the test. Maybe we can get around the issue using an `aria-label`.

Reply:
You know...I think we have Cypress Testing Library. Funny enough, I forgot about it. If the `Button` component has the `role` attribute set to `button`, we could even do something like this:

```
cy.findByRole('button', {name: /Create cluster registration secret/i}).click()
```

Reply:
It doesn't have the button role. 😞 

It does have the PatternFly prop for OUIA testing, but I'm not sure whether or not we should start relying on those, or if that is really a semantic property that makes sense to be the primary way we select elements: `data-ouia-component-type="PF5/Button"`.